apps/pipelines/models.py

```diff
@@ -499,6 +519,28 @@ def requires_attachment_tool(self) -> bool:
         """When a collection is linked, the attachment tool is required"""
         return self.params.get("collection_id") is not None
 
+    def _archive_related_params(self):
+        """
+        Archive related params that were also versioned along with this node
+        """
+        from apps.assistants.models import OpenAiAssistant
+        from apps.documents.models import Collection
+        from apps.pipelines.nodes import nodes
+
+        model_param_specs = {
+            nodes.AssistantNode.__name__: [ModelParamSpec(param_name="assistant_id", model_cls=OpenAiAssistant)],
+            nodes.LLMResponseWithPrompt.__name__: [
+                ModelParamSpec(param_name="collection_id", model_cls=Collection)
+                # TODO: Custom actions needed
+            ],
+        }
+
+        for spec in model_param_specs.get(self.type, []):
+            if instance_id := self.params[spec["param_name"]]:
+                instance_cls = spec["model_cls"]
+                obj = instance_cls.objects.get(id=instance_id)
```

Comment:
Could be refactored to `spec.get_object(sepf.params)`

Should probably also add error handling with logging in case the object doesn't exist.

Reply:
https://github.com/dimagi/open-chat-studio/pull/1405/commits/1b4d3a27b4dd576f83c1ee756dc0cbe00db851d7